file-events/src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -22,11 +22,28 @@ public static String getVersion() {
 
     private static native String getVersion0();
 
+    /**
+     * Returns if in the current environment file events are supported.
+     *
+     * Currently on Linux file events are only supported on glibc.
+     * Specifically on musl libc it is not supported.
+     */
+    public abstract boolean isSupported();
+
+    private void ensureSupported() {
+        if (!isSupported()) {
+            throw new NativeException("File events are not supported");
+        }
+    }
+
     /**
      * Forces the native backend to drop the cached JUL log level and thus
      * re-query it the next time it tries to log something to the Java side.
+     *
+     * @throws NativeException if file events are not supported.
      */
     public void invalidateLogLevelCache() {
+        ensureSupported();
```

Comment:
I think this should be better done in `FileEvents.getEventFunctions()`, so you can't even get the functions when glibc is not supported.

Reply:
Good point.